lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -348,11 +350,23 @@ def extract_details(exam_attempt, features):
                 u'{status} Comments'.format(status=status): '; '.join(comment_list),
             })
 
+        proctored_exam['track'] = get_course_mode(exam_attempt['user_id'], exam_attempt['course_id'])
         return proctored_exam
 
     exam_attempts = get_exam_violation_report(course_key)
     return [extract_details(exam_attempt, features) for exam_attempt in exam_attempts]
 
+def get_course_mode(pstudent_id, pcourse_id):
+    """
+     Returns course mode for a course/student
+     """
+    course_key = CourseKey.from_string(pcourse_id)
+
+    coursemode = CourseEnrollment.objects.filter(user_id=pstudent_id, course_id=course_key)
```

Comment:
Prefer the classmethod: `CourseEnrollment.get_enrollment(user, course_key)`

Better yet, load all enrollments using one query and apply them to the results returned from `get_exam_violation_report`.

Reply:
Updated based on some of your comments. Let me know if I am on the right path.

Reply:
I'm pretty sure we're looking at something like https://docs.djangoproject.com/en/2.1/ref/models/querysets/#select-related (or something similar) to ensure we don't have to issue a query to the database for each of the already-serialized rows from the attempt table.